src/sqlast/ddl.rs

```diff
@@ -84,3 +79,106 @@ impl ToString for TableConstraint {
         }
     }
 }
+
+/// SQL column definition
+#[derive(Debug, Clone, PartialEq, Hash)]
+pub struct SQLColumnDef {
+    pub name: SQLIdent,
+    pub data_type: SQLType,
+    pub collation: Option<SQLObjectName>,
+    pub constraints: Vec<ColumnConstraint>,
+}
+
+impl ToString for SQLColumnDef {
+    fn to_string(&self) -> String {
+        format!(
+            "{} {}{}",
+            self.name,
+            self.data_type.to_string(),
+            self.constraints
+                .iter()
+                .map(|c| format!(" {}", c.to_string()))
+                .collect::<Vec<_>>()
+                .join("")
+        )
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Hash)]
+pub enum ColumnConstraint {
```

Comment:
The current AST seems to be partially modeled after Postgres in which any column option (except COLLATE) is called a "column constraint", even though it doesn't constrain anything. Shall we add "GENERATED" to _this_ struct when it's implemented? Other options, like MS-specific ones? Would we keep the `ColumnConstraint` name then? If we plan to rename, it might be better to rename it before landing.

Reply:
Good points. I've pretty substantially reworked the design of the AST. I've renamed the main enum to `ColumnOption`, to better reflect the fact that it might not contain only constraints. I've also moved the optional name to a wrapping `ColumnOptionDef` struct, which better allows *any* of the options to take on a name, to allow for Postgres syntax, where practically everything can have a name. 

I think this should provide the right amount of future flexibility. GENERATED naturally fits as a new enum variant in `ColumnOption`, and hopefully it's less grating now that the struct doesn't refer to "constraints." MSSQL-specific options can also go into `ColumnOption`, if that makes sense, or they can live as fields on `SQLColumnDef`, if they're the kind of options (like SPARSE) that can't be named and must appear in a fixed order.